### PR TITLE
390 - push notifications for admins (frontend)

### LINKED ISSUE #390: Task 3 - Push Notifications for administrators
Short-term closures should be 24 hours. (Or even a little less than 24 hours, when email job runs)
Long-term closures should be day of expiration.

Tasks:
- Add a date-picker component for selecting end of longterm closure.
- Add option for an indefinite closure.
- Change backend schema to handle durations differently.
- Change all status_duration queries on frontend
- Create a cron job to gather all closed crossings and send a single email notification to admins.
- get austintexas.gov email address

### PR DESCRIPTION
**Pull request by Niicck**
resolves #390

### CODE REVIEW

File: deployment/devDeployConfig.js [hunk 1]
```diff
@@ -5,9 +5,9 @@
 
   [branch-name]: {
       deploy: Boolean (indicate whether to deploy or not),
-      backend: Service Endpoint for backend you want to point to.
+      backend: Branch name of the backend you want to point to.
```
Comment (briaguya) [inline deployment/devDeployConfig.js]: 😁😁😁

File: deployment/writeDevBackend.js [hunk 2]
```diff
@@ -1,22 +1,42 @@
...
+      console.log(`No serviceEndpoint output for stack ${stackName}. ${advice}`);
+      process.exit(1);
+    }
+    const serviceEndpointValue = serviceEndpointOutput[0].OutputValue;
+    if (!serviceEndpointValue) {
+      console.log(`No serviceEndpoint value for stack ${stackName}. ${advice}`);
+      process.exit(1);
+    }
+    if (serviceEndpointValue) {
+      return fs.writeFileSync(`${__dirname}/dev_react_app_backend_url.tmp`, `export REACT_APP_BACKEND_URL=${serviceEndpointValue}`);
```
Comment (briaguya) [inline deployment/writeDevBackend.js]: Just curious, why are we using a file here?

File: src/components/Dashboard/ManageUsersPage/EditUser.js [hunk 3]
```diff
@@ -31,7 +31,7 @@ class EditUser extends Component {
         firstName: '',
         lastName: '',
         role: 'floods_community_editor',
-        communityId: props.currentUser.communityId,
+        communityId: (props.currentUser.role === 'floods_super_admin' ? '9001' : props.currentUser.communityId),
```
Comment (briaguya) [inline src/components/Dashboard/ManageUsersPage/EditUser.js]: I see how this works but we should probably put a low priority issue in the backlog to make sure we don't always need to have a community with an ID of 9001 for super admins to work (not going to be an issue with our current setup but might cause issues in the future if we try using different datasets etc.)

File: src/components/Shared/Emoji/index.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+// https://medium.com/@seanmcp/%EF%B8%8F-how-to-use-emojis-in-react-d23bbf608bf7
+const Emoji = props => (
```
Comment (briaguya) [inline src/components/Shared/Emoji/index.js]: At first I was curious as to if there was a lib on npm that already did this but my quick googling didn't lead to a solution using one being as clean as this. Great way to manage errors! (probably going to steal this component for Janis)

File: src/services/apolloClientService.js [hunk 5]
```diff
@@ -13,12 +13,17 @@ const httpLink = createHttpLink({
...
 const errorLink = onError(({ graphQLErrors, networkError, operation, forward }) => {
   // TODO: make an error page for displaying all errors
-  logError("graphQLErrors", graphQLErrors);
-  logError("networkError", networkError);
-  if ( graphQLErrors.name === "JsonWebTokenError") {
-    localStorage.removeItem('jwt_user_token');
-    window.location.reload(); // Refreshing page will redirect unauthenticated user to the login page.
+  if (graphQLErrors) {
+    graphQLErrors.forEach((err) => {
+      if (err.name === "JsonWebTokenError") {
```
Comment (briaguya) [inline src/services/apolloClientService.js]: 👍 👍 👍

File: deployment/writeDevBackend.js (hunk 2 above)
Comment (Niicck) [reply deployment/writeDevBackend.js]: That's the only way (I've found) to propagate env variable information from js to serverless. 

js -> file -> sourced into bash env => build serverless deployment

**Pull request merged**